editor/gui/editor_dir_dialog.cpp

```diff
@@ -43,16 +44,35 @@ void EditorDirDialog::_update_dir(TreeItem *p_item, EditorFileSystemDirectory *p
 
 	p_item->set_metadata(0, path);
 	p_item->set_icon(0, tree->get_editor_theme_icon(SNAME("Folder")));
-	p_item->set_icon_modulate(0, tree->get_theme_color(SNAME("folder_icon_color"), SNAME("FileDialog")));
+
+	const Color default_folder_color = tree->get_theme_color(SNAME("folder_icon_color"), SNAME("FileDialog"));
 
 	if (!p_item->get_parent()) {
 		p_item->set_text(0, "res://");
+		p_item->set_icon_modulate(0, default_folder_color);
 	} else {
 		if (!opened_paths.has(path) && (p_select_path.is_empty() || !p_select_path.begins_with(path))) {
 			p_item->set_collapsed(true);
 		}
 
 		p_item->set_text(0, p_dir->get_name());
+
+		Dictionary assigned_folder_colors = FileSystemDock::get_singleton()->get_assigned_folder_colors();
+		if (assigned_folder_colors.has(path)) {
+			bool editor_is_dark_theme = FileSystemDock::get_singleton()->get_editor_is_dark_theme();
+			const Color &folder_color = FileSystemDock::get_singleton()->get_folder_colors()[assigned_folder_colors[path]];
```

Comment:
those three values could be set once when the dialogue is made visible as they won't change. In addition dark theme should call directly `EditorThemeManager::is_dark_theme` instead of relying on `FileSystemDock`.